### PR TITLE
Don't `#include <ostream>` outside of test-only code.

### PR DESCRIPTION
**Pull request by bjacob**
On Android/ARM64, Release config, size of `iree-run-module` (after stripping):

Before: 794424
After: 591472

### CODE REVIEW

File: runtime/src/iree/tooling/vm_util.cc [hunk 1]
```diff
@@ -245,44 +244,42 @@ Status ParseToVariantList(iree_hal_allocator_t* device_allocator,
   return OkStatus();
 }
 
+#define IREE_PRINTVARIANTLIST_CASE_I(S, B, V)                                 \
+  case IREE_VM_VALUE_TYPE_I##S:                                               \
+    iree_string_builder_append_format(B, "i" #S "=%" PRIi##S "\n", (V).i##S); \
```
Comment (benvanik) [inline runtime/src/iree/tooling/vm_util.cc:249]: all the append methods may fail and will need to be wrapped in IREE_RETURN_IF_ERROR

**Review by benvanik — CHANGES_REQUESTED** [review]
otherwise lgtm!

File: runtime/src/iree/tooling/vm_util.cc (hunk 1 above)
Comment (bjacob) [reply runtime/src/iree/tooling/vm_util.cc]: done

**Review by benvanik — APPROVED** [review]

**Comment by bjacob** [comment]
@benvanik substantial conflict with #10433 which expanded usage of `<ostream>` around this area, PTAL again.

**Review by benvanik — APPROVED** [review]

**Pull request merged**